scripts/run-plugin-nuts.js

```diff
@@ -0,0 +1,206 @@
+  const results = execSync(
+    `npm show ${module.name}@${module.version} ${['time', 'version', ...sections].join(' ')} --json`
+  );
+  let jsonResults = JSON.parse(results);
+  if (isArray(jsonResults)) {
+    jsonResults = jsonResults
+      .sort((a, b) => new Date(b.time[b.version]).getTime() - new Date(a.time[a.version]).getTime())
+      .find((result) => new Date(result.time[result.version]).getTime() <= timeCreated.getTime());
+  }
+  return jsonResults;
+};
+
+const getOrgAndProject = (module) => {
+  const results = execSync(`npm show ${module.name} homepage`);
+  if (Reflect.ownKeys(manualRepoOverrides).includes(module.name)) {
+    return manualRepoOverrides[module.name];
+  }
+  const repoUrl = new URL(results.toString().trim());
+  const [, org, repo] = repoUrl.pathname.split('/');
+  return { org, repo };
+};
+
+const orgSlug = (org) => {
+  return `github/${org}`;
+};
+const projectSlug = (org, repo) => {
+  return `${orgSlug(org)}/${repo}`;
+};
+
+/**
+ * Triggers the just-nuts workflow for the named org/repo/branch
+ * @param sfdxVersion
+ * @param branch
+ * @returns {Promise<*>}
+ */
+const triggerNutsForProject = async (sfdxVersion, module, branch = 'main') => {
+  const body = {
+    branch: branch,
+    parameters: {
+      'run-auto-workflows': false,
+      'run-just-nuts': true,
+      sfdx_version: sfdxVersion,
+      repo_tag: module.version,
+    },
+  };
+  return await circle(`${circleciBaseUrl}project/${projectSlug(module.info.org, module.info.repo)}/pipeline`, {
+    method: 'POST',
+    json: body,
+  });
+};
+
+/**
+ * Triggers the nuts monitoring workflow
+ * @param jobData
+ * @param branch
+ * @returns {Promise<*>}
+ */
+const triggerNutsMonitor = async (jobData, branch = 'main') => {
+  const body = {
+    branch: branch,
+    parameters: {
+      'run-auto-workflows': false,
+      'run-just-nuts': true,
+      nuts_job_data: JSON.stringify(jobData),
+    },
+  };
+  return await circle(`${circleciBaseUrl}project/${projectSlug('salesforcecli', 'sfdx-cli')}/pipeline`, {
+    method: 'POST',
+    json: body,
+  });
+};
+/**
+ * Makes a request to named URL. Ensures that auth token is applied.
+ * @param url
+ * @param options
+ * @returns {Promise<any>}
+ */
+const circle = async (url, options = {}) => {
+  const defaultOptions = {
+    method: 'GET',
+    headers: { 'Circle-Token': process.env.CIRCLECI_API_TOKEN, responseType: 'json', resolveBodyOnly: true },
+  };
+  options = { ...defaultOptions, ...options };
+  const response = await (async () => {
+    try {
+      return await got(url, options);
+    } catch (error) {
+      console.log(error);
+      throw new Error(error && error.response?.body ? error.response.body : error);
+    }
+  })();
+  return JSON.parse(response.body);
+};
+
+/**
+ * Examines a lit of plugins to determine which have NUT tests
+ * @returns {*}
+ */
+const qualifyPluginsWithNonUnitTests = (timeCreated, modules) => {
```

Comment:
I may not understand what this is trying to do.  Is there a reason not to look into the yarn.lock file to see what sfdx-cli resolved the plugin versions to be (since that's what it's going to build/ship) rather than working with npm, timestamps, etc?

Reply:
you could probably do the same `.filter((module) => /^@*salesforce/.test(module.name));` on that?

Reply:
The yarn.lock is an interesting idea. I like that better than the timestamp approach. Would that work @peternhale ?